README.md

```diff
@@ -1,20 +1,241 @@
+The panicking way was chosen, as the `Result`s seemed less ergonomic for potential developers and they feel a bit inconsistent
+as `CallResult` already has `CanisterError` and `CanisterReject` variants. Also going with the panics and having the state reverted
+felt more familiar with the smart contracts on other blockchains.
+
+## Implementation notes
+
+### Helios
+
+The foundation of this canister. The ethereum canister depends on the `client`, `consensus`, `execution`, and `common` helios crates.
+To be able to use a helios on the ICP it had [to be forked](https://github.com/eigerco/helios). The fork introduced many
+changes to the internals of helios thus making itself not-upstreamable in its current form. Probably it is possible to upstream
+some of those changes and maybe even make it compatible with the ICP ecosystem but this is considered to be not a trivial task
+with many possible ways of achieving the goals that should be considered and brainstormed.
+The changes were made in a manner that helios still works perfectly fine for native, but when targeting wasm it only supports
+the ICP and no longer the browsers.
+
+The changes include:
+- updating the `helios` and making it compatible with the rust `stable` toolchain
+- getting rid of any occurrences of `wasm-bindgen`, `glue-timers`, and other browser-related wasm dependencies
+- getting rid of `tokio`-related stuff that is incompatible with wasm
+- using `ethers-core` where possible and removing `ethers-providers` entirely
+- replace `reqwest` with https outcalls when targeting wasm
+- update the `revm` to v3 and introduce a way to fetch missing slots outside in an async manner
+- change the updating logic to only lock for a short period and add proper shutdown and cleanup
+
+For the complete list of changes see this [comparison](https://github.com/a16z/helios/compare/master...eigerco:helios:master).
+
+### Ethers
+
+The `ethers` crate is the building block of helios and the home for most of the types from the Ethereum ecosystem. It also
+provides the utilities to generate types and encoding logic from the smart contract's ABI. It would be really useful to have this
+crate working for the ICP ecosystem completely nonetheless this is considered a nontrivial task. The main problem is that it
+has already good support for browser-side wasm and some of its subcrates utilize not supported crates heavily. The main pain
+point we've encountered was the `ethers-providers` crate. The best way to approach this would be to support ICP in crates
+like [`instant`](https://github.com/sebcrozet/instant) and [`futures-timer`](https://github.com/async-rs/futures-timer) or even
+[`gloo-timers`](https://github.com/rustwasm/gloo), guard the http implementation behind the feature flag and add a way for a user
+to add a custom provider.
+
+There was one change made to the ethers, allowing the use of `ethers-contract` without the need to depend on `ethers-providers`
```

Comment:
I think we should mention that our team did the change

Reply:
I have avoided using first-person pronouns like 'we' or 'our team' for the whole document. Do you maybe have an idea how could we make it clearer that we made that change without using 'we'? or am I too pedantic there and we should just use 'we'?

Reply:
Ok keep it like this. Maybe we can mention it in the weekly update or something.